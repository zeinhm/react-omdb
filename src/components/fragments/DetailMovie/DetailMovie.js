import React from 'react'
import { useSelector } from 'react-redux'
import { PropagateLoader } from "react-spinners";
import defaultImg from '../../../assets/default-poster.jpg'
import './styles.scoped.css'

export default function DetailMovie() {
  const state = useSelector( state => state.movieDetail)
  const defaultPoster = (e) => {
    e.target.src = defaultImg
  }
  return (
    <main className='detail-card'>
     {(state?.dataMovieDetail && !state?.isLoadingMovieDetail) &&
        (
          <div className='container'>
            <div className='poster'>
              <img alt="Poster" src={state.dataMovieDetail.Poster} 
                onError={defaultPoster}/>
            </div>
            <div className='detail'>
              <h2>{state.dataMovieDetail.Title}</h2>
              <p>{state.dataMovieDetail.Plot}</p>
              <h3>{state.dataMovieDetail.Genre}</h3>
              <div className='more-info'>
                <div>
                  <span>Original Release:</span>
                  <h4>{state.dataMovieDetail.Released}</h4>
                </div>
                <div>
                  <span>Ratings:</span>
                  <h4>{state.dataMovieDetail.Ratings?.[0]?.Value}</h4>
                </div>
                <div>
                  <span>Runtime:</span>
                  <h4>{state.dataMovieDetail.Runtime}</h4>
                </div>
                <div>
                  <span>Box Office:</span>
                  <h4>{state.dataMovieDetail.BoxOffice}</h4>
                </div>
              </div>
              <div className='actors'>
                <h3>Actors:</h3>
                <span>{state.dataMovieDetail.Actors}</span>
              </div>
            </div>
          </div>
        )
      } 
      <a href="http://www.omdbapi.com/" target="_blank">Powered by OMDb API</a>
      <div className='loading-wrapper'>
        <PropagateLoader
          loading={state?.isLoadingMovieDetail}
          color='#d16ba5'
          />
      </div>
    </main>
  )
}
