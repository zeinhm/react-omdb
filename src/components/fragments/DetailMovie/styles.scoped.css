.detail-card{
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.container, .poster, .poster img {
  border-radius: 20px 0;
}

.container {
  display: flex;
  background-color: white;
  margin-top: 5rem;
  width: 1024px;
}

.poster {
  width: 670px;
  height: 500px;
}

.poster img {
  height: 100%;
  width: 100%;
  object-fit: cover;
  object-position: 50% 50%;
}

.detail {
  padding: 1.5rem 2rem;
}

.detail p {
  margin: 2rem 0
}

.more-info  {
  align-items: center;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

.more-info div {
  margin: 2rem;
}

.actors h3 {
  margin-bottom: 1rem;
}

.detail-card > a {
  margin: 1rem auto;
}

.loading-wrapper {
  align-items: center;
  display: flex;
  justify-content: center;
  margin: 5rem;
}

@media(max-width: 1024px) {
  .container {
    display: flex;
    flex-direction: column;
    width: 80%;
  }

  .poster {
    width: 100%;
    height: 300px;
  }
  .more-info div {
    margin: 1rem;
  }
}

@media(max-width: 425px) {
  .container {
    display: flex;
    flex-direction: column;
    width: 80%;
  }
  /* .poster img {
    width: 100%;
  } */
  .more-info div {
    margin: 1rem;
  }
}

