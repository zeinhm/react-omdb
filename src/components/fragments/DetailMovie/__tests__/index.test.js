import React from 'react';
import { useSelector } from 'react-redux';
import ShallowRenderer from 'react-test-renderer/shallow';
import DetailMovie from '../DetailMovie';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(cb => {
    cb({});
    return {
      state: {
        dataMovieDetail: {},
        isLoadingMovieDetail: false
      }
    };
  }),
}));

describe('src/components/fragments/DetailMovie', ()=> {
  test('DetailMovie', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<DetailMovie />);
    
    useSelector.mockImplementationOnce(() => ({
      state: {
        dataMovieDetail: {},
        isLoadingMovieDetail: false
      }}));
    expect(tree).toMatchSnapshot();
  });
});
