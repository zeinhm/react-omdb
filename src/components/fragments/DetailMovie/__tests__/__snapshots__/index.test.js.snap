// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/fragments/DetailMovie DetailMovie 1`] = `
<main
  className="detail-card"
>
  <a
    href="http://www.omdbapi.com/"
    target="_blank"
  >
    Powered by OMDb API
  </a>
  <div
    className="loading-wrapper"
  >
    <Loader
      color="#d16ba5"
      css=""
      loading={true}
      size={15}
    />
  </div>
</main>
`;
